### ISSUE TITLE
Parser not working when used with Vanilla Extract

### ISSUE THREAD

**Issue by Qocotzxin** [Ephem/rollup-plugin-preserve-directives#16]
Hi, first of all thanks for the amazing work, this helped me a lot!

I don't think this is actually a bug, but it might help someone and also, I wanted to ask if the way I solved it is the correct one.

I'm using [Vanilla Extract](https://vanilla-extract.style/), React and Vite to build a component library. While trying to add support for RSC I encountered the issue that this library fixes, but the problem is that VE generates CSS files during build time, so when running the transform phase, specifically when calling the Acorn instance (`const ast = this.parse(code);`) it throws an error because it can't parse a css.(cjs/js) file (this is the part I don't really understand why. Thought it made sense but checking in detail each file, it should be able to parse it I guess?). My simple and naive solution was to just avoid parsing these files by adding the following code in the transform hook:

```
      if (/.\.css$/.test(id)) {
        return { code, map: null };
      }
```

If there is a better solution and/or this is a change that could be applied to this package please let me know, else I hope this helps someone in the same situation!

Thanks again for this plugin!

**Comment by Ephem** [comment]
Sorry for not responding to this earlier, I've been busy and out of touch with this plugin for a while.

I think this might be related to the order of the plugins running. If `preserveDirectives` is placed above the Vanilla Extract plugin, could you try reordering them so Vanilla Extract runs first and see if that solves it?

**Comment by Ephem** [comment]
#20 should solve this by ignoring files that end with .css, it's released in 0.3.1.

I'll go ahead and close this, get in touch again if this didn't solve it for you! 😄

**Comment by cetchebarne** [comment]
Hey @Ephem thanks for getting back. It's been a while, but that's exactly how I ended up resolving it on my end, thanks!
